server/aesdsocket.c

```diff
@@ -138,74 +307,59 @@ int main(int argc, char *argv[])
+        current_thread_data->id = thread_id;
+        SLIST_INSERT_HEAD(&head, current_thread_data, next_ptr);
+        curr_ptr = SLIST_FIRST(&head);
+        while (curr_ptr != NULL)
         {
-            total_bytes += bytes_recv;
-            // make a valid string
-            temp_buffer[total_bytes] = '\0';
-            // check for a new line
-            if (strchr(temp_buffer, '\n') != NULL)
-                break;
-            // if not then we need to increase buffer_size
-            buffer_size += buffer_size;
-            char *new_buffer = realloc(temp_buffer, buffer_size);
-            if (new_buffer == NULL)
+            itr_ptr = SLIST_NEXT(curr_ptr, next_ptr); // Store the next item before potential removal
+            if (curr_ptr->is_thread_complete)
             {
-                syslog(LOG_ERR, "Failed to extend buffer size!");
-                free(temp_buffer);
-                close(aux_fd_write);
-                close(new_fd);
-                return FAILURE;
+                // Join the completed thread
+                if (pthread_join(curr_ptr->id, NULL) != 0)
+                {
+                    syslog(LOG_ERR, "Unable to join thread: %s", strerror(errno));
+                }
+                else
+                {
+                    syslog(LOG_INFO, "Thread joined successfully");
+                }
+
+                // Remove from the list
+                SLIST_REMOVE(&head, curr_ptr, thread_node, next_ptr);
+                syslog(LOG_INFO, "Removing thread resources");
+
+                // Free memory and prevent dangling pointer
+                free(curr_ptr);
             }
-            temp_buffer = new_buffer;
-        }
-        // new line happened
-        write(aux_fd_write, temp_buffer, total_bytes);
-        close(aux_fd_write);
 
-        // once a new line is appear we need to loop back to client
-        int aux_fd_read = open(FILE_LOG_PATH, O_RDONLY);
-        if (aux_fd_read == -1)
-        {
-            syslog(LOG_ERR, "Failed to open file to read");
-            free(temp_buffer);
-            close(new_fd);
-            continue;
+            // Move to the next item
+            curr_ptr = itr_ptr;
         }
-        lseek(aux_fd_read, 0, SEEK_SET);
-        int bytes_read;
-        do
-        {
-            bytes_read = read(aux_fd_read, temp_buffer, buffer_size);
-            if (bytes_read > 0)
-                if (send(new_fd, temp_buffer, bytes_read, 0) == -1)
-                {
-                    syslog(LOG_ERR, "Error sending data to client");
-                    break;
-                }
-        } while (bytes_read > 0);
-        syslog(LOG_ERR, "closing file after send");
-        free(temp_buffer);
-        close(aux_fd_read);
-        close(new_fd);
-        syslog(LOG_INFO, "Closed connection from %s", client_ip);
     }
+    pthread_join(timestamp_thread, NULL);
```

Comment:
This won't work, would hang forever.  You are currently exiting due to the exit(0) in the signal handler and this cleanup is dead code.  Should result in valgrind warnings/errors.